server/backends/chunkstore/chunkstore.go

```diff
@@ -0,0 +1,454 @@
+package chunkstore
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"math"
+	"syscall"
+	"time"
+
+	"github.com/buildbuddy-io/buildbuddy/server/interfaces"
+	"github.com/buildbuddy-io/buildbuddy/server/util/status"
+)
+
+const (
+	mb   = 1 << 20
+	year = time.Hour * 24 * 365
+)
+
+// This implements a chunking reader/writer interface on top of an arbitrary
+// blobstore, averting the need to access blobs all at once.
+type Chunkstore struct {
+	internalBlobstore    interfaces.Blobstore
+	writeBlockSize       int
+	writeTimeoutDuration time.Duration
+}
+
+type ChunkstoreOptions struct {
+	WriteBlockSize       int
+	WriteTimeoutDuration time.Duration
+}
+
+func New(blobstore interfaces.Blobstore, co *ChunkstoreOptions) *Chunkstore {
+	writeBlockSize := co.WriteBlockSize
+	if writeBlockSize == 0 {
+		writeBlockSize = 1 * mb
```

Comment:
for these defaults, it'd be nice to define const defaults at the top of the file and use those if the value is not specd